docs/json-api.md

```diff
@@ -1544,10 +1545,47 @@ curl -X GET "http://localhost:3689/api/library/tracks/1"
   "data_kind": "file",
   "path": "/music/srv/Incubus/Make Yourself/12 Pardon Me.mp3",
   "uri": "library:track:1",
-  "artwork_url": "/artwork/item/1"
+  "artwork_url": "/artwork/item/1",
+  "lyrics": 1
 }
 ```
 
+### Get lyrics for track
+
+Get lyrics for a specific track in your library
+
+**Endpoint**
+
+```http
+GET /api/library/lyrics/{id}
```

Comment:
Thanks for reviewing!

The artwork is a specific API too, since the Track links to an Artwork URL (not the artwork itself). Lyrics are also optional (like artwork) and can be quite large. I'm not sure it's a good idea to push (optional) large data to the user, unless she actually wants it.   

In the current web implementation, it doesn't call the track endpoint currently (it's doing fine with the queue endpoint), so using the track endpoint to store the lyrics means transferring a lot of useless metadata when only the lyrics are requested.

Reply:
Ok, but if size a concern then it might also not be great to have it as a mfi and dbmfi attribute and an extra column in the db? Then it should be made like artwork, i.e. with lyrics_url and treating the lyrics as a separate resource.

Reply:
I can't say. It's an architectural decision here. Lyrics are (specially formatted) text, in the order of 1kB to 10kB for the largest one I would say. There is no processing to perform on them like for artwork (no resizing, the parsing of lyrics to synchronized sentences is done in the web client upload loading). 

So on one side, if we have to store them like artwork, it would mean extracting them and storing as LRC files or txt file or re-parsing them on the fly from the MP3 file when requested and dealing with a cache like artwork is dealt with. That's a lot of work, IMHO.

On the other side, unlike artwork because they are encoded as text, they can be stored in the DB while the files are parsed for metadata. That's a relatively small overload (typically for a 1000 song database, it increases the DB by 5MB I would say). That's less than a single song size. This saves latency because you don't have to re-launch ffmpeg to re-parse the file for metadata extraction each time it's required. 

In the implementation, I've tried to avoid changing anything to existing users, so they don't have to pay for this feature if they don't use it. By simply adding a `lyrics` key in the track JSON object **only if** lyrics were detected, it's the smallest impact on them they could have. Then, only users that have enabled the lyrics panel will fetch the /api/library/lyrics and fetch the additional data. 

If we were to move to a lyrics dynamic extraction URL like artwork, it would mean one query per track (+ one server parsing the song per track) even if it doesn't have lyrics. I think it's not interesting.

Currently, the `/api/library/lyrics` *is* the Lyrics URL, but it's fetched from cached parsing done while we had parsed the MP3 files and saved in the DB. It's not dynamic like artwork because it wouldn't make sense, we don't ask for downsized lyrics.  

I hope it's a bit clearer why I did it this way. Please let me know your opinion on this.

Reply:
> 1kB to 10kB

That sounds quite small, so in that light I think it should just be a metadata attribute.

Reply:
Ok, done in the 2nd commit, now "lyrics"'s key of the track object is filled with lyrics if present. I've updated the documentation to remove the lyric route and the web application to use track instead.